torch/csrc/jit/codegen/cuda/lower_shift.cpp

```diff
@@ -415,16 +604,16 @@ void HaloInfo::build(TensorDomain* td) {
         " of ",
         td->getRootDomain());
 
-    if (halo_width == 0) {
-      halo_width_map_.insert({root_axis, 0});
+    if (!halo_info.hasHalo()) {
```

Comment:
Nothing particular. It was just because it needed additional overloaded versions of `makeAddExpr`, so I just didn't bother to use it. 

Expression simplification is necessary for shift validation, e.g., when we want to prove `<symbolic-val-X> + 1` and `<symbolic-val-X> + (0 + 1)` are the same. It may make more sense to make these functions as part of the KIR builder.